lib/stat_calculator.rb

```diff
@@ -171,4 +171,68 @@ def worst_offense
     worst_offense = @teams.find { |team| team.team_id.to_s == worst_offense_team.to_s}
     worst_offense.team_name
   end
+
```

Comment:
Suggestions: Both methods iterate through @teams to find the matching team after determining the team ID.
Solution: Use a hash (team_lookup) to map team_id to team_name for constant-time lookups.

```ruby
team_lookup = @teams.each_with_object({}) do |team, hash|
  hash[team.team_id.to_s] = team.team_name
end
```
**Then, replace the final loop with:**

```ruby
team_lookup[team_with_most_tackles.to_s]
```